### PR TITLE
Fix mypy errors & enable checks for backend

### PR DESCRIPTION
**Pull request by joel-wright**
Enhance type checking to improve code quality:
- Fix existing type errors in backend code
- Enable mypy checking on backend code in pre-commit hook
- Update schema in places where the current types do not express the intent (e.g. optional types for non-optional fields)

### CODE REVIEW

File: backend/app/api/api_v1/routers/actions.py [hunk 1]
```diff
@@ -65,18 +65,14 @@ async def action_create(
...
     )
 
     try:
         db_action = create_action(db, action_create)
     except Exception as e:
         if isinstance(e, IntegrityError):
-            raise HTTPException(
-                409, detail=f"Database integrity error, underlying={e.orig}"
-            )
+            raise HTTPException(409, detail=f"Database integrity error, underlying={e}")
```
Comment (opyate) [inline backend/app/api/api_v1/routers/actions.py:75]: Might be useful to know what [orig](https://docs.sqlalchemy.org/en/14/core/exceptions.html#sqlalchemy.exc.DBAPIError) is?

> The wrapped exception object is available in the [StatementError.orig](https://docs.sqlalchemy.org/en/14/core/exceptions.html#sqlalchemy.exc.StatementError.orig) attribute. Its type and properties are DB-API implementation specific.

File: backend/.pre-commit-config.yaml [hunk 2]
```diff
@@ -23,5 +23,6 @@ repos:
     rev: v0.931
     hooks:
       - id: mypy
+        files: ^backend/
```
Comment (opyate) [inline backend/.pre-commit-config.yaml:26]: I guess we add `loader` and `pipeline` later?

File: backend/app/api/api_v1/routers/actions.py (hunk 1 above)
Comment (opyate) [reply backend/app/api/api_v1/routers/actions.py:75]: Actually, this doesn't matter for now. We need to revise the error-handling at some point 👍🏻

File: backend/.pre-commit-config.yaml (hunk 2 above)
Comment (joel-wright) [reply backend/.pre-commit-config.yaml:26]: yeah, we'll have to, they have a good chunk of errors themselves

File: backend/app/api/api_v1/routers/actions.py (hunk 1 above)
Comment (joel-wright) [reply backend/app/api/api_v1/routers/actions.py:53]: I spent quite a while trying to get the type checker to accept that `.orig` but it stubbornly refused to believe it existed. The other option is to just disable the type check on that line...

**Review by opyate — APPROVED** [review]

**Pull request merged**